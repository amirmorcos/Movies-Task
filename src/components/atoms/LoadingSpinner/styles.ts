import { StyleSheet } from "react-native";
import { Colors } from "themes/Colors";

export default StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors.BLACK,
    justifyContent: "center",
    alignItems: "center",
  },
});
