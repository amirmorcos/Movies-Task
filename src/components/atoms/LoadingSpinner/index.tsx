import React from "react";
import { ActivityIndicator, View } from "react-native";
import styles from "./styles";

const LoadingSpinner = () => {
  return (
    <View style={styles.container}>
      <ActivityIndicator />
    </View>
  );
};

export default LoadingSpinner;
