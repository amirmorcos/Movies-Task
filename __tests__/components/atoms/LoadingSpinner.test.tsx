import LoadingSpinner from "atoms/LoadingSpinner";
import React from "react";
import renderer from "react-test-renderer";

jest.mock("react-native-vector-icons/FontAwesome6", () => "Icon");
jest.mock("react-native-size-matters", () => ({
  s: jest.fn(),
  vs: jest.fn(),
  moderateScale: jest.fn(),
}));

test("renders correctly", () => {
  const tree = renderer.create(<LoadingSpinner />).toJSON();
  expect(tree).toMatchSnapshot();
});
