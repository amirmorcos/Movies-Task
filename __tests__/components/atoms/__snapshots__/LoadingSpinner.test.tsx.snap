// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "black",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator />
</View>
`;
